### DISCUSSION TITLE
ZEEK and MODBUS (additional tcp ports: 501, 503 y 504)

### DISCUSSION THREAD

**Discussion by kornflake003-GITHUB** [Security-Onion-Solutions/securityonion#6634] [Unsupported Versions]
Hello, where can I tell the ZEEK that for the MODBUS protocol it also captures information for the destination tcp ports 501, 503 and 504. Currently it only captures information for the tcp port 502.

![ZEEK Support](https://user-images.githubusercontent.com/85261291/146623169-f718bf0c-7f84-4ee8-afab-f9f118468bdb.png)

¿Or maybe the configuration has to be in steno?

**Comment by kornflake003-GITHUB** [comment]
I already verified that if we have traffic on TCP ports 501 and 503, with TCPDUMP.

**Reply by tomjones1977** [reply]
According to the Zeek docs the Dynamic Protocl Detection works regardless of the port in use.  See https://docs.zeek.org/en/master/logs/dpd.html for more info.  You may be able to look at dpd.log in the zeek logs to work out why it's not recognising modbus on other ports as modbus protocol.

**Reply by kornflake003-GITHUB** [reply]
Thanks for your support @tomjones1977 , i´m already check the zeek logs at: /nsm/zeek/logs/current, and it´s contain a lot of information about connections on tcp ports: 501,502 and 503. I think the problem is on HUNT or KIBANA modules, because the "event.dataset" field for tcp ports (501 & 503) is set to "conn", not to "modbus" as it for TCP port 502. ¿Any suggestion? 

Somewhere in ZEEK I have to indicate that the data flow of TCP ports 501 & 502 is MODBUS too.

**Reply by dougburks** [reply]
The problem is not with Hunt or Kibana. If Zeek recognized the traffic as modbus, it would record the connection to both the conn log and the modbus log. As @tomjones1977 mentioned, you should check the dpd.log for additional clues as to why Zeek is not recognizing modbus. You might also check the weird log for additional clues.
